lib/cli/src/Cardano/CLI.hs

```diff
@@ -1045,22 +1047,39 @@ cmdWalletUpdatePassphrase mkClient =
                     "Please enter your current passphrase: "
                 wPassphraseNew <- getPassphraseWithConfirm
                     "Please enter a new passphrase: "
+                let passwds =
+                        OldNewPassword wPassphraseOld wPassphraseNew
                 runClient wPort (const mempty) $
                     putWalletPassphrase mkClient (ApiT wId) $
-                        WalletPutPassphraseData
-                            (ApiT wPassphraseOld)
-                            (ApiT wPassphraseNew)
+                        producePassphraseData passwds
             Left _ ->
                 handleResponse Aeson.encodePretty res
 
+data OldNewPassword passphraseUpdate =
+    OldNewPassword (Passphrase "raw") (Passphrase "raw")
+
+class PassphraseUpdate passphraseUpdate where
+    producePassphraseData :: OldNewPassword passphraseUpdate -> passphraseUpdate
+
+instance PassphraseUpdate ByronWalletPutPassphraseData where
+    producePassphraseData (OldNewPassword old new) =
+        if old == Passphrase @"raw" "" then
+            ByronWalletPutPassphraseData Nothing (ApiT new)
+        else
+            ByronWalletPutPassphraseData (Just $ ApiT old) (ApiT new)
+
+instance PassphraseUpdate WalletPutPassphraseData where
+    producePassphraseData (OldNewPassword old new) =
+        WalletPutPassphraseData (ApiT old) (ApiT new)
```

Comment:
The extra parameter and instances above bring quite some overhead. I'd suggest to leave the command as it is. It won't be possible to update legacy wallets that have no or a invalid passphrase set via the command-line. That's an acceptable limitation :+1:

Reply:
reverted support for CLI